pkg/monitor/cluster/prometheusalerts.go

```diff
@@ -106,14 +108,23 @@ func (mon *Monitor) emitPrometheusAlerts(ctx context.Context) error {
 	return nil
 }
 
-func alertIsIgnored(alertName string) bool {
-	if strings.HasPrefix(alertName, "UsingDeprecatedAPI") {
+func alertIsIgnored(alert model.Alert) bool {
+	if strings.HasPrefix(alert.Name(), "UsingDeprecatedAPI") {
 		return true
 	}
 
-	if _, ok := ignoredAlerts[alertName]; ok {
-		return true
-	}
+	if options, ok := ignoredAlerts[alert.Name()]; ok {
+		// alert suppression rule is not namespace bound
+		if options.namespaces == nil {
+			return true
+		}
 
+		// alert suppression rule is namespace-bound
+		for _, ns := range options.namespaces {
+			if ns == string(alert.Labels["namespace"]) {
```

Comment:
Now that I think of it, probably it would be better to have a map to store the ignored namespaces (and match them quickly) rather than a list to iterate on (perf wise, if the list of the list is long)

Reply:
updated